charts/backstage/templates/_helpers.tpl

```diff
@@ -21,3 +21,29 @@ Returns a secret name for service to service auth
         {{- include "common.names.fullname" . -}}-auth
     {{- end -}}
 {{- end -}}
+
+{{/*
+Sets the secretKeyRef name for Backstage to the PostgreSQL existing secret if it present
+*/}}
+{{- define "janus-idp.postgresql.secretName" -}}
+    {{- if ((((.Values).global).postgresql).auth).existingSecret -}}
+        {{- .Values.global.postgresql.auth.existingSecret -}}
+    {{- else if .Values.postgresql.auth.existingSecret -}}
+        {{- .Values.postgresql.auth.existingSecret -}}
+    {{- else -}}
+        {{- printf "%s-%s" .Release.Name "postgresql" -}}
+    {{- end -}}
+{{- end -}}
+
+{{/*
+Get the password secret.
```

Comment:
Can you please link to the reference where we copypasted it from? So we know this is not our template actually.

Reply:
Add the line: Referenced from: https://github.com/bitnami/charts/blob/main/bitnami/postgresql/templates/_helpers.tpl#L94-L105